cpp/arcticdb/version/version_utils.hpp

```diff
@@ -230,3 +230,40 @@ void fix_stream_ids_of_index_keys(
     const StreamId &stream_id,
     const std::shared_ptr<VersionMapEntry> &entry);
 }
+
+inline arcticdb::proto::descriptors::SortedValue deduce_sorted(arcticdb::proto::descriptors::SortedValue existing_frame, arcticdb::proto::descriptors::SortedValue input_frame){
```

Comment:
Isn't this switch equivalent to,

```
if (input_frame == existing_frame) {
  return input_frame;
}

if ((input_frame == UNKNOWN || existing_frame == UNKNOWN) 
  && (input_frame != UNSORTED || existing_frame != UNSORTED)) {
  return UNKNOWN;
}

return UNSORTED;
```

Reply:
yes agreed. But I didn't want to do any logic simplification here. Because for 2 reasons: 

1. We want the state machine to be as  more clear as possible.
2. Improbable, but if in the future we want to do changes to this state machine, that could need more changes in the logic, rather that just changing some values in the current state machine logic.

Reply:
Agree with point 2. But still not convinced for 1.